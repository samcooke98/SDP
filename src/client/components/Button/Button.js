import React from "react";
import styles from "./web.css";

export default class Button extends React.Component {
    render() {
        var styleClass = styles.baseBtn
        styleClass += " " + styles[this.props.variant]

        return (
            <button 
                onClick={this.props.onClick} 
                style={ Object.assign(
                    {
                        display: 'block', marginLeft: "auto", marginRight: 'auto', 
                        width: this.props.width, 
                        height: this.props.height, 
                        backgroundColor: this.props.colour || ''
                    }, this.props.style) }
                className={ styleClass }
            >
                <p> {this.props.label}</p>
            </button>
        )

    }

    someFunc = () => {
        //'This' is automatically bound when writing functions like this
        return JSON.stringify(this.props);
    }
}

Button.defaultProps  = { 
    variant: "primary"
}

//In addition to CSS modules, you may wish to use: https://github.com/gajus/react-css-modules