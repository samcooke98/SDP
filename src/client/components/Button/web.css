/* Button Reset */

button {
    background-color: #f0f0f0;
    border: 1px solid rgb(0, 0, 0);
    border: 1px solid rgba(0, 0, 0, 0.1);
    line-height: 1em;
    margin: 0.25em;
    box-sizing: border-box;
    display: inline-block;
    display:flex;
    align-items: center;

    font-family: "Raleway";

    cursor: pointer;
}

button > * { margin: 0; }

.baseBtn {
    text-transform: uppercase;
    border-radius: 100px;
    margin: 0
}

.primary {
    background-color: #2F80ED;
    color: #F8F8F8;
    letter-spacing: 3.6px;
}

.secondary {
    background-color: #F8F8F8;
    color: #2F80ED;
    letter-spacing: 3.6px;
}


.clear {
    background-color: rgba(0, 0, 0, 0);
    border-color: #000000;
    color: #000000;
}

button h1 {
    margin: 0;
}

button h3 {}