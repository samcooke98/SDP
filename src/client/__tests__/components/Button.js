import React from "react";
import renderer from "react-test-renderer";
// import ReactDOM from 'react-dom';

import Button from "../../components/Button/Button";


describe('React Component: Button', () => {
    it("matches a snapshot", () => {
        const tree = renderer.create(
            <Button />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})