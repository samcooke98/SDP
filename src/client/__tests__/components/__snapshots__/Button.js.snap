// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Component: Button matches a snapshot 1`] = `
<button
  className="undefined undefined"
  onClick={undefined}
  style={
    Object {
      "backgroundColor": "",
      "cursor": "pointer",
      "display": "block",
      "height": undefined,
      "marginLeft": "auto",
      "marginRight": "auto",
      "width": undefined,
    }
  }
>
  <p
    style={
      Object {
        "fontSize": "24px",
      }
    }
  >
     
  </p>
</button>
`;
